lib/s3middleware/objectRetention.js

```diff
@@ -0,0 +1,125 @@
+const { parseString } = require('xml2js');
+
+const errors = require('../errors');
+
+/*
+    Format of xml request:
+    <Retention>
+      <Mode>COMPLIANCE|GOVERNANCE</Mode>
+      <RetainUntilDate>2020-05-20T04:58:45.413000Z</RetainUntilDate>
+    </Retention>
+*/
+
+function validateMode(mode) {
+    const validMode = {};
```

Comment:
Calling these variables valid* is a bit misleading as sometimes they are just errors. Maybe a more generic term like "results" or something that doesn't imply a state?